tests/backup/backup_restore_basic_test.go

```diff
@@ -128,20 +128,16 @@ var _ = Describe("{BasicSelectiveRestore}", func() {
 		opts[SkipClusterScopedObjects] = true
 		log.InfoD("Deleting deployed applications")
 		ValidateAndDestroy(contexts, opts)
-
 		backupDriver := Inst().Backup
 		backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
 		log.FailOnError(err, "Failed while trying to get backup UID for - [%s]", backupName)
-
 		log.InfoD("Deleting backup")
 		_, err = DeleteBackup(backupName, backupUID, orgID, ctx)
```

Comment:
It seems that the step is unnecessary since CleanupCloudSettingsAndClusters will handle it. Additionally, @sn-px may already be taking care of it. Also, note that DeleteBackup does not wait until the backup is fully deleted. If you believe that CleanupCloudSettingsAndClusters cannot delete the backup, you may need to add a status check.